src/lib.rs

```diff
@@ -77,18 +79,26 @@ impl MsgEnvelope {
     /// So, needs some improvement..
     pub fn verify(&self) -> Result<bool> {
         let data = if self.proxies.is_empty() {
-            bincode::serialize(&self.message)
-                .map_err(|_| Error::Serialisation("Could not serialize message".to_string()))?
+            self.serialise()?
         } else {
             let mut msg = self.clone();
             let _ = msg.proxies.pop();
-            bincode::serialize(&msg)
-                .map_err(|_| Error::Serialisation("Could not serialize message".to_string()))?
+            msg.serialise()?
         };
         let sender = self.most_recent_sender();
         Ok(sender.verify(&data))
     }
 
+    pub fn from(bytes: &[u8]) -> Result<Self> {
+        let wire_msg = WireMsg::from(bytes)?;
+        wire_msg.deserialise()
+    }
+
+    pub fn serialise(&self) -> Result<Vec<u8>> {
+        let wire_msg = WireMsg::new(self, PayloadSerType::Json)?;
```

Comment:
`Ok(WireMsg::new(self, PayloadSerType::Json)?.serialise())`
also, if WireMsg is to be used like this, shall we have a function like `WireMsg::serialise(self, PayloadSerType::Json)` ?

Reply:
Yeah, I'll add such a function.